### ISSUE TITLE
(T5) Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer

### ISSUE THREAD

**Issue by dreamingjudith** [dreamingjudith/paper-reading-list#1]
URLs
- [Arxiv](https://arxiv.org/abs/1910.10683)

# Vanilla Transformer와의 가장 큰 차이점

- Layer normalization의 Bias를 제거
- Layer normalization을 Residual path 바깥에 배치
- Relative positional embedding을 사용함

# C4 - The Colossal Clean Crawled Corpus

- CommonCrawl의 정제된 버전으로 파이썬의 `langdetect` 패키지를 이용해 영어만 걸러냄

# Training

- 인코더, 디코더 각각 12층
- $d_{ff}$: 3072, $d_{kv}$: 64, $d_{model}$: 768
- Number of attention heads: 12
- ReLU activation
- Dropout probability: 0.1
- Around 65,536 tokens for each batch
    - Max. sequence length: 512
    - Batch size: 128 sequences
    - 한 배치 안에 문장을 꽉꽉 채워넣는 방식으로 구성함 (tensor2tensor의 기본 배칭 방식)
- Sees around 34B ($2^{35}$) tokens during pre-training
- "Inverse square root" learning rate scheduler
    - 첫 $10^4$ step 동안은 0.01로 고정
    - Then exponentially decays the learning rate
- 524,288 ($2^{19}$) steps
- 32K개의 WordPiece vocabulary

# Training objective

![image](https://user-images.githubusercontent.com/20284375/134338451-8e152143-42d2-4d44-bacf-c3794e85f361.png)

- 연속된*Consecutive* span 단위로 토큰을 마스킹한 다음 Sentinel token으로 대체
    - Original BERT와 마찬가지로 입력 문장의 15%를 마스킹
    - 위 그림 예제에서는 **\<X\>**, **\<Y\>**, **\<Z\>**가 Sentinel token을 나타냄
- 타겟은 오로지 Sentinel token에 들어가는 Consecutive span이 뭐였는지만 맞추게 되는 방식
    - 예제에서는 Sentinel token의 끝을 **\<Z\>**로 표현함
